src/clip.sol

```diff
@@ -328,6 +328,17 @@ contract Clipper {
         return active[idx];
     }
 
+    // Returns boolean for if an auction needs a redo (reset)
+    function needsRedo(uint256 id) external returns (bool) {
+        // Read auction data
+        Sale memory sale = sales[id];
+
+        // Compute current price [ray]
+        uint256 price = calc.price(sale.top, sub(now, sale.tic));
+
+        return (sale.tab > 0 && (sub(now, sale.tic) > tail || rdiv(price, sale.top) < cusp));
```

Comment:
WDYT about taking this line and turning it into an `internal` function which can be reused in all three places it occurs? This may slightly increase gas costs for `take` and `redo`, but since storage calls dominate gas costs anyway, IMO it's worth the readability/robustness.

Reply:
Ill do a gas analysis on this today.